### PR TITLE
improvement: add tests for business logic (part 1)(SDKCF-1880)

### PR DESCRIPTION
**Pull request by imaginaris**

### CODE REVIEW

**Comment by imaginaris** [comment]
@jz709u I can't add you as a reviewer...

File: RInAppMessaging/Classes/Services/ConfigurationService.swift [hunk 1]
```diff
@@ -50,18 +51,17 @@ extension ConfigurationService {
...
+            let sdkVersion = bundleInfo.inAppSdkVersion else {
 
                 CommonUtility.debugPrint("InAppMessaging: failed creating a request body")
                 assertionFailure()
                 return .failure(RequestError.unknown)
         }
 
         let getConfigRequest = GetConfigRequest(
-            locale: locale,
+            locale: Locale.current.identifier,
```
Comment (donnie-jp) [inline RInAppMessaging/Classes/Services/ConfigurationService.swift]: Is this change supported by the backend?

Comment (imaginaris) [reply RInAppMessaging/Classes/Services/ConfigurationService.swift]: According to the documentation this parameter is optional and there is no format requirement. I don't know why this extension was created, the standard `identifier` should do the work. I added a question to the list for backend team to answer and all changes that will arise from their answers will be added in other task. (Requests from the app are still passing)

File: Tests/UIColorExtensionsTests.swift [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+                }
+
+                it("will return color with requested alpha value") {
+                    let color = UIColor(fromHexString: "000000", alpha: 0.7)
+                    var alpha: CGFloat = 0
+                    color?.getRed(nil, green: nil, blue: nil, alpha: &alpha)
+                    expect(alpha).to(equal(0.7))
+                }
+
+                it("will return full black color with FFFFFF") {
```
Comment (donnie-jp) [inline Tests/UIColorExtensionsTests.swift]: black -> white

File: Tests/UIColorExtensionsTests.swift [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+                    let color = UIColor(fromHexString: "FFFFFF")
+                    expect(color).to(equal(UIColor.whiteRGB))
+                }
+
+                it("will return full white color with 000000") {
```
Comment (donnie-jp) [inline Tests/UIColorExtensionsTests.swift]: white -> black

**Pull request merged**